channels/serializers.py

```diff
@@ -80,9 +82,45 @@ def get_membership_is_managed(self, channel):
         """
         Get membership_is_managed from the associated Channel model
         """
-        channel_obj = Channel.objects.get(name=channel.display_name)
+        channel_obj = self._get_channel(name=channel.display_name)
         return channel_obj.membership_is_managed
 
+    def get_avatar(self, channel):
+        """Get the avatar image URL"""
+        channel_obj = self._get_channel(name=channel.display_name)
+        try:
+            return channel_obj.avatar.url
+        except ValueError:
+            return None
+
+    def get_banner(self, channel):
+        """Get the banner image URL"""
+        channel_obj = self._get_channel(name=channel.display_name)
+        try:
+            return channel_obj.banner.url
+        except ValueError:
+            return None
+
+    def validate_avatar(self, value):
+        """Empty validation function, but this is required for WriteableSerializerMethodField"""
+        if not hasattr(value, 'name'):
+            raise ValidationError("Expected avatar to be a file")
+        return {"avatar": value}
+
+    def validate_banner(self, value):
+        """Empty validation function, but this is required for WriteableSerializerMethodField"""
+        if not hasattr(value, 'name'):
+            raise ValidationError("Expected banner to be a file")
+        return {"banner": value}
+
+    def _get_channel(self, name):
```

Comment:
Related to other comment, what is the benefit of having `channels` in the context and trying to get the channel from there first instead of just calling `Channel.objects.get(name=name)`?

Reply:
The `channels` is there to cache `Channel` objects so we can fetch them all at once and reuse them rather than fetching once per field or channel.

The `try` `catch` here is because of an edge case where we are creating a new channel. The `Channel` object isn't created until some point within the `create` method. Since the context from `get_serializer_context` is created before `create` is invoked there won't be any channels in the object.